cob_hardware_emulation/src/emulation_clock.cpp

```diff
@@ -0,0 +1,61 @@
+#include <ctime>
+#include <boost/asio.hpp>
+#include <boost/bind.hpp>
+#include <boost/date_time/posix_time/posix_time.hpp>
+
+#include <ros/ros.h>
+#include <rosgraph_msgs/Clock.h>
+
+
+class EmulationClock
+{
+public:
+    EmulationClock(boost::asio::io_service& io, int dt_ms, double time_factor)
+        : dt_ms_(dt_ms),
+          time_factor_(time_factor),
+          timer_(io, boost::posix_time::milliseconds(dt_ms))
+    {
+        t_ = (double)std::time(NULL);
+        pub_ = nh_.advertise<rosgraph_msgs::Clock>("/clock", 1);
+        timer_.async_wait(boost::bind(&EmulationClock::timer_cb, this));
+    }
+
+    void timer_cb()
+    {
+        t_ += time_factor_*dt_ms_/1000;
```

Comment:
Some `()` would be good here maybe. 
Not that the calculation changes, but I'm not sure what happens with the double/int conversions going on under the hood.